### PR TITLE
More pr comments

### PR DESCRIPTION
**Pull request by Tenaria**
Changes:
* They updated the otel stuff to v0.14 (after pulling in master) which broke a lot of things so I had to make a lot of changes around these
* Moved go.mod one level up as per their suggestion "Will we ever need to instrument things from the github.com/aws/aws-sdk-go itself? If so, would it make sense to have this module match that structure by being one level up?"
* Changed file structure to match naming standards. Now is `go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go/service/s3/otels3`
* Moved helper for `AppendSpanAndTraceIDFromSpan` into `otels3` package
* changed instrumentationName variable to be package name, not instrumented package name
* changed `defer span.End()` to capture exact end timestamp of aws sdk function call
* Set span status and added failure/success attributes to span in s3 instrumentation

tldr addressed all the comments that are unresolved here https://github.com/open-telemetry/opentelemetry-go-contrib/pull/416 and additionally merged master in an fixed all the conflicts.

### CODE REVIEW

**Comment by hstan** [comment]
merge base with upstream to make the PR changeset smaller?

**Comment by Tenaria** [comment]
@hstan yeah sure, just didn't want to put anything through that hasn't been reviewed but it should be fine if it's just pulling in master. I'll do that now.

**Comment by Tenaria** [comment]
done

File: instrumentation/github.com/aws/aws-sdk-go/service/s3/otels3/s3.go [hunk 1]
```diff
@@ -64,18 +64,23 @@ func (s *instrumentedS3) PutObjectWithContext(ctx aws.Context, input *s3.PutObje
...
-	defer span.End()
 
 	output, err := s.S3API.PutObjectWithContext(ctx, input, opts...)
+	callReturnTime := trace.WithTimestamp(time.Now())
+	defer span.End(callReturnTime)
 
 	if err != nil {
 		attrs = append(attrs, labelStatusFailure)
+		span.SetAttributes(labelStatusFailure)
+		span.SetStatus(codes.Ok, err.Error())
```
Comment (chenzhihao) [inline instrumentation/github.com/aws/aws-sdk-go/service/s3/otels3/s3.go]: is that expected to use `codes.ok` even there is an error?

Comment (Tenaria) [reply instrumentation/github.com/aws/aws-sdk-go/service/s3/otels3/s3.go]: ah that's a mistake, thanks for picking up will fix

**Review by chenzhihao — APPROVED** [review]

**Comment by Tenaria** [comment]
Changes:
